### PR TITLE
migration(variables,datasets): add shortName and version to variables and datasets tables

### PR DESCRIPTION
**Pull request by Marigold**
We need `shortName` and `version` for better alignment of ETL and grapher metadata. See [accompanying PR](https://github.com/owid/etl/pull/359)

### CODE REVIEW

File: db/migration/1659011799966-AddShortNameAndVersionToDatasets.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            ADD COLUMN shortName VARCHAR(255)
+            DEFAULT NULL;
+        `)
+        await queryRunner.query(`
+            ALTER TABLE datasets
+            ADD COLUMN version VARCHAR(255)
+            DEFAULT NULL;
+        `)
+        await queryRunner.query(`
+            ALTER TABLE datasets ADD UNIQUE (shortName);
```
Comment (larsyencken) [inline db/migration/1659011799966-AddShortNameAndVersionToDatasets.ts:18]:
```suggestion
            ALTER TABLE datasets ADD CONSTRAINT unique_short_name_version_namespace UNIQUE (shortName, version, namespace);
```

File: db/migration/1659011799966-AddShortNameAndVersionToDatasets.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        `)
+    }
+
+    public async down(queryRunner: QueryRunner): Promise<void> {
```
Comment (larsyencken) [inline db/migration/1659011799966-AddShortNameAndVersionToDatasets.ts:22]:
```suggestion
    public async down(queryRunner: QueryRunner): Promise<void> {
        await queryRunner.query(`
            ALTER TABLE datasets DROP INDEX unique_short_name_version_namespace;
        `)
```

File: db/migration/1659011791860-AddShortNameToVariables.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    implements MigrationInterface
+{
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`
+            ALTER TABLE variables
+            ADD COLUMN shortName VARCHAR(255)
+            DEFAULT NULL;
+        `)
+        await queryRunner.query(`
+            ALTER TABLE variables ADD UNIQUE (shortName);
```
Comment (larsyencken) [inline db/migration/1659011791860-AddShortNameToVariables.ts:13]:
```suggestion
            ALTER TABLE variables ADD CONSTRAINT unique_short_name_per_dataset (shortName, datasetId);
```

File: db/migration/1659011791860-AddShortNameToVariables.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        `)
+    }
+
+    public async down(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`
```
Comment (larsyencken) [inline db/migration/1659011791860-AddShortNameToVariables.ts:18]:
```suggestion
        await queryRunner.query(`
            ALTER TABLE variables
            DROP INDEX unique_short_name_per_dataset
        `)
        await queryRunner.query(`
```

**Review by larsyencken — CHANGES_REQUESTED** [review]
The constraints are too aggressive, since short names can be reused in different namespaces. I think it should be:

- On `variables`: each `(short_name, datasetId)` should be unique, ignoring nulls
- On `datasets`: each `(short_name, version, namespace)` should be unique, ignoring nulls

**Comment by Marigold** [comment]
Added `shortName` and `version` update for existing ETL datasets.

**Review by larsyencken — APPROVED** [review]
Looks good to go! Great that you included the data changes we want in the migration too.

**Pull request merged**